src/shims/env.rs

```diff
@@ -465,4 +465,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(())
     }
+
+    fn getpid(&mut self) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        let target_os = &this.tcx.sess.target.os;
+        assert!(
+            target_os == "linux" || target_os == "macos",
+            "`getpid` is only available for the UNIX target family"
+        );
+
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("getpid", reject_with)?;
+            this.set_last_error_from_io_error(ErrorKind::PermissionDenied)?;
+
+            return Ok(-1);
+        }
+        // The reason we need to do this wacky of a conversion is because
+        // `libc::getpid` returns an i32, however, `std::process::id()` return an u32,
+        // So we need to fit that u32 into an i32 and pray
+        if std::process::id() > i32::MAX.try_into().unwrap() {
+            throw_unsup_format!("Process id width bigger than i32")
+        } else {
+            Ok(std::process::id().try_into().unwrap())
+        }
```

Comment:
This is not idiomatic Rust -- you should call `try_into` and then handle the `None` case, rather than doing the check twice.

You can use something like `try_into().ok_or_else(|| err_unsup_format!(...))`, see e.g. [here](https://github.com/rust-lang/miri/blob/1fa63f1e9b490fb33ff38781a7af57c36987a392/src/shims/os_str.rs#L35). Or you can use `if let Some(id) = std::process::id().try_into()`. Whatever you prefer. But no `unwrap` please. :)

Reply:
> This is not idiomatic Rust -- you should call `try_into` and then handle the `None` case, rather than doing the check twice.
> 
> You can use something like `try_into().ok_or_else(|| err_unsup_format!(...))`, see e.g. [here](https://github.com/rust-lang/miri/blob/1fa63f1e9b490fb33ff38781a7af57c36987a392/src/shims/os_str.rs#L35). Or you can use `if let Some(id) = std::process::id().try_into()`. Whatever you prefer. But no `unwrap` please. :)

I couldn't figure out what I should put in the body of the `if let`, and also, `ok_or_else` seem to be an `Option` thing

Reply:
Ah right, `try_into` returns a `Result`; in that case use `map_err`.

Reply:
But also see https://github.com/rust-lang/miri/pull/2215#discussion_r892712741, maybe we should just wrap around the value as I assume that is what the Linux implementation of `std::process::id()` does anyway.

Reply:
> Ah right, `try_into` returns a `Result`; in that case use `map_err`.

I don't understand how to use this function, see zulip